### DISCUSSION TITLE
Quarkus build failing after updating to 2.16.0.Final version

### DISCUSSION THREAD

**Discussion by adisnuhan** [quarkusio/quarkus#30784] [Community]
`Hello` community, 

we upgraded this week to latest quarkus version (2.16.0.Final) and we noticed that our builds are failing with very curious exception (see execption bellow). So it seems that the quarkus-maven-plugin is missing some parameter which we never had in the plugin configuration. This is our current plugin configration:

![image](https://user-images.githubusercontent.com/13799369/216073431-4c705e0a-b283-47a0-91af-130fc60332f9.png)

The application is built with "mvn clean install" command. If we revert the version back to 2.15.3.Final then it works without any problems. 

I also checked the migration manuals (from 12.15.3.Final -> 12.16.0.Final) and was not able to find there aynthing relevant. 

What parameter is missing here? Can someone help us? 

![image](https://user-images.githubusercontent.com/13799369/216073553-6154209b-62fc-4fcd-8cb9-a1708bfcc82f.png)

**Comment by adisnuhan** [comment]
After l invested a lot of time investigating here is the cause and the solution (maybe workaround) for this problem. 

So the error ocurrs in this line of quarkus itself (not my project) : https://github.com/quarkusio/quarkus/blob/main/core/deployment/src/main/java/io/quarkus/deployment/steps/ProfileBuildStep.java
and further following the classes I found that the default profile is not set when building quarkus:
https://github.com/quarkusio/quarkus/blob/main/core/runtime/src/main/java/io/quarkus/runtime/LaunchMode.java

So to solve this i provided the default profile via mvn command line mvn -U clean install -Dquarkus.profile=prod

This is from my point of view a defect because the quarkus should set this value on its own if nothing else is provided. I will create an defect in github repository

**Reply by gsmet** [reply]
@adisnuhan thanks for digging into this. Could you provide a small reproducer in your issue? Because if it was a common problem, I would have expected a lot of reports so I suspect something specific to your setup somehow. Thanks!
